### DISCUSSION TITLE
Problem communicating with arduino output serial port

### DISCUSSION THREAD

**Discussion by Mymi-INS** [bonsai-rx/bonsai#1167] [Q&A] [answered]
Hello everyone, 
 I have an issue ! I try to read the serial.print of my arduino with bonsai SerialStringRead but i'm not succeeding to do it... Here is my bonsai configuration. 

![image](https://user-images.githubusercontent.com/61882756/205107236-2de5da0d-28a2-4e7f-914f-4b77d19a7296.png)

If i remove the "my arduino" node I can start the code but nothing appear in the CSV but with it I have an error, something like : bonsai don't have access to the port of my arduino. Can you help with that?

here is the file : 
[Monmaze_question (1).zip](https://github.com/bonsai-rx/bonsai/files/10134232/Monmaze_question.1.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya,
Try to move the `CreateArduino` to the top of the `GroupWorkflow`. Additionally, change the `PortName` property from a specific port to the name of the Arduino object (in your case `myArduino`).

Hope it helps :)

**Comment by Mymi-INS** [comment]
Hi Bruno!!

Thanks a lot! Your help totally solved my issue :D 

Have a great day!!
